src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/FeatureSwitchManager.cs

```diff
@@ -572,6 +570,13 @@ private bool TryGetConstantArgument(MethodIL methodIL, byte[] body, OpcodeFlags[
                             constant = (int)substitution.Value;
                             return true;
                         }
+                        else if (method.IsIntrinsic && method.Name is "op_Inequality" or "op_Equality"
+                            && method.OwningType is MetadataType mdType
+                            && mdType.Name == "Type" && mdType.Namespace == "System" && mdType.Module == mdType.Context.SystemModule
+                            && TryExpandTypeEquality(methodIL, body, flags, currentOffset, method.Name, out constant))
```

Comment:
Would it also take disjoint case into account with `Type` stored in a local? 👀

Roslyn has propensity to emit [such code](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGADAAQY4B0ASgK4B2ALgJYC2MlAwhKwA7MAbGFADKwgG7MwMAM4BuQoQDE9WgIEBDYENIATZjK1DCGAMzkspToQDehUg9IB6J6QBUpADKyZAFQAWGvQ4pEG6Xj4BQZaB4jCkABQaAqwQMowAlKTMujBMUsmkAJKe9o4ujmGkvFAQurTS2bn5YIWQuQDmeZRlDhUe3jJ+gfTmVYPDQSiksfFJKWmZTXksrQLFnokwCNK8jKQAZtCkjP4wvc6uDsAQHbQypKrqYGdgANbZ9BFDUfQoWVUanUGvEcisCut2jAuvQADQXfrfSb0ACsMw0D1O8XozAOjAAnqQZPimBoELDDhABG8HhA4lATmdPoxhPQYIwHrBIKx2PRwsxGAirnpaPFGBBshyiQI2GyhhT8ez0ZimQB3aAyeJQmGKAiOUgAbQAsuz/HUinwBAkTadzZaAPJ7ZgQegySgAQQ6HVgQ2YcSK9BlOPoHQyAF0LmZyDgkKQblSkb8cAAeXwAaRg+IAfAl05nSEI8RS84SoMwOv5Mhc7Hr9Y5cYkCbwYBADrmM/isgBeLsnfHN1sJYD4lkZLIYADsiWHo4SEGAACsYGBMoX9snpyOYBk54vl5kyxXGApa3XsgdG/2W22S93e03rwkZDPt+Op0+Xzv50uVxk16QNw/Lcvz3X9D0rE8zwcBsEgfQdb1IHs+wHNt7jNKBMjfRI0OgTJdx/VcYDxADsJkdC8O/fcMnA48Ln1GC4JvDs72Qx8yNwsdyHfdiMJAgi/yI9dEh4ijQIPcsILo+sL1gq94OYxD7zk1DmCYTjJ2w1TRP4/9ANoLS+KomjIKghjlPbTMWMYhIDKwmy1PwqjdMSAzHLAiTaNPOszJQizO0U1jBzUF1Qzs4KQ0M39nIScLQzc8SjxMs8fMfBCkOsgQQvU99Moi+KBOIwDcriyj3MSqSHA0g5kk1JLSAAX11fVjVNO1eCtG0zV0C12sdFgXTdT1vR8P0YADINVNDCMvKjChY3jdYJl+LBUw7HMSwLQTiw7UgaIyCrEKzQKmMsolVQFF4DprKC6wAfVIVUzi+ay0qU3zP0OzdZ1KwjCq+7d8po+EvJu+7HryY6/Ks8zn2Az6gO+sSCqEhGAZ+6iPOBm79TBp7IdeyGcN4+Gie0pzBJImKRMihLKyx7GHFxiGXoU9KYfI7sjqfDn8ui6nAcxg6cYevGWdOtnfP0tSSdc9HoqlsmyrpoXHCZ57zIJ6yDPh2Wkei3X+KBlXGZF5mNdZt7H1iznsOKmnkcp62BaPemGbV/GLchu34bt3mKaKrLneVkGoPursjuqgRNQO+qTwuFrbW6y1rVapPeqdAaPS9H0ZFG8bVMm8NI3MOa4wgBMlpGUxVszdadrXbb8z26sDt8K9SA+S35Msur6Jkzuu7bT87M/P2/qHYCg88m6YIHyHYdHOyF7RvX/eE0f0eMg7Z4C6zSey0ieblteqaPpGt5D6D+939mOKXs+dJP/nN483vpMSOe99srjNIc4/x4VvbC+M9r4S0fN/DS9lFa/RRgbIyr9t6gMHjFO2YVfb/xRk7F+5VL7ng/jfXyqCf4JHQavcepDDYINwVVGqMA6qNQIPHTqbUOqpx6gIPqzpXRZ2Gr6f0gYC4RWmvqWaMYy4V0iCMFANdsx+U2kWUgG1m64PDh7U6MhzqMEurg66DNFHt32ODZ6BDHwfVURPRGj9x4b3PoLXB+o27NhOKbYxYDBzLxtqje2fMbGUJdsbfRTjDGiwCoTamJNn5kMwZEvxwc9GBLFC45xbi2zhPMTE8m48MlK0YK7bGjjElGOScgwBMs/5RMdnAnJeSboFOcUU/YKSoGeKqTAymrSMb+PsY4OpwTmYmKCt7cxWCKl6QofArp8TelJMacgoZXNxlRRPos2muSAlhwjrQmOccvIJy6uwlOid2GcMzkNHOecBHBimsXaM81y6LUkUEFEMi675gboonayibqqNyNVNQjAoZnQuv4K6ASXzw18ZklGkLqkBOXvDDxY8UaIuwXEvRpMIkPyhY7bJqyalQREgirFSysnErxQEhWZToEOz0h0o23ToJMB1uUqxsCWUTLRQzWKJMVk0ttoHVFayGUFhCj7AVozEi8vpfEjZhwtm4NjoQeqQA) with some modern patterns (see `LessThan4()`).

Reply:
It will work for sdloc immediately followed by ldloc, but otherwise this algorithm is pretty garbage (it was written mostly just to match what IL Linker can propagate). The north star for this code is to be deleted and replaced by RyuJIT (#83021).